### ISSUE TITLE
Add dropout to neural estimators

### ISSUE THREAD

**Issue by pawel-czyz** [cbg-ethz/bmi#133]
MINE, InfoNCE, and other neural estimators could use random state to use dropout. This requires changes in the training loop and some refactoring.
